src/main/java/io/opensaber/views/Transformer.java

```diff
@@ -3,51 +3,48 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.node.JsonNodeFactory;
 import com.fasterxml.jackson.databind.node.ObjectNode;
-
 import java.util.ArrayList;
 import java.util.List;
 
 public class Transformer {
-
+    /**
+     * transforms a given JsonNode to representation of view templates
+     * view template indicates any new field or mask fields for transformation
+     * 
+     * @param viewTemplate
+     * @param node
+     * @return
+     */
     public JsonNode transform(ViewTemplate viewTemplate, ObjectNode node) {
         ObjectNode result = JsonNodeFactory.instance.objectNode();
         String subjectType = node.fieldNames().next();
-
         ObjectNode nodeAttrs = (ObjectNode) node.get(subjectType);
-        // for each field array item
+
         for (Field field : viewTemplate.getFields()) {
-            // - if function is specified
-            /// - call inline function expression evaluation
-            if (field.getFunction() != null) {
-                // TODO - remove hardcoded function reference
-                String expression = viewTemplate.getFunctionDefinitions().get(0).getResult();
-                String functionStr = field.getFunction();
-                String argNames = field.getFunction().substring(functionStr.indexOf("(") + 1,
-                                                            functionStr.lastIndexOf(")"));
-                String[] functionArgs = argNames.split(", ");
+
+            String functionStr = field.getFunction();
+            if (functionStr != null) {
+
+                String fdName = field.getFunctioName();
+                String expression = viewTemplate.getExpression(fdName);
 
                 FieldFunction function = new FieldFunction(expression);
                 List<Object> actualValues = new ArrayList<>();
-                for(String oneArg: functionArgs) {
+                for (String oneArg : field.getArgNames()) {
                     // Cut off the $
-                    //System.out.println("Added value "+ oneArg.substring(1) + " -- " + nodeAttrs.get(oneArg.substring(1)));
                     actualValues.add(ValueType.getValue(nodeAttrs.get(oneArg.substring(1))));
                 }
                 function.setArgValues(actualValues);
 
                 FunctionEvaluator<String> evaluator = new FunctionEvaluator(function);
 
                 if (field.getDisplay()) {
-                    result.put(field.getTitle(), evaluator.evaluate());
+                    result.put(field.getTitle(), evaluator.evaluate().toString());
```

Comment:
The toString here can still make 7 as "7", which is why we needed some other way.

Reply:
Now, JsonNodeFactory.instance.pojoNode(evaluator.evaluate()) using to set the object.